File: appmy/src/main/java/com/leilei/guoshujinfu/mylearning/BaseApplication.java
package com.leilei.guoshujinfu.mylearning;

import android.app.Activity;
import android.app.Application;
import android.os.Bundle;

import com.facebook.drawee.backends.pipeline.Fresco;
import com.leilei.guoshujinfu.mylearning.util.AppCache;
import com.leilei.guoshujinfu.mylearning.util.AppLog;

/**
 * @Author: AqrLei
 * @Name MyLearning
 * @Description:
 * @Date: 2017/8/7
 */


public class BaseApplication extends Application implements Application.ActivityLifecycleCallbacks {
    @Override
    public void onCreate() {
        super.onCreate();
        registerActivityLifecycleCallbacks(this);
        Fresco.initialize(this);
        AppCache.init(this);
        AppCache.setFileName("MyLearning");
    }

    @Override
    public void onActivityCreated(Activity activity, Bundle bundle) {
        AppLog.logDebug(AppLog.LOG_TAG_ACTIVITY, activity.getClass().getSimpleName() + "Created");

    }

    @Override
    public void onActivityStarted(Activity activity) {
        AppLog.logDebug(AppLog.LOG_TAG_ACTIVITY, activity.getClass().getSimpleName() + "Started");

    }

    @Override
    public void onActivityResumed(Activity activity) {
        AppLog.logDebug(AppLog.LOG_TAG_ACTIVITY, activity.getClass().getSimpleName() + "Resumed");

    }

    @Override
    public void onActivityPaused(Activity activity) {
        AppLog.logDebug(AppLog.LOG_TAG_ACTIVITY, activity.getClass().getSimpleName() + "Paused");
    }

    @Override
    public void onActivityStopped(Activity activity) {
        AppLog.logDebug(AppLog.LOG_TAG_ACTIVITY, activity.getClass().getSimpleName() + "Stopped");
    }

    @Override
    public void onActivitySaveInstanceState(Activity activity, Bundle bundle) {
        AppLog.logDebug(AppLog.LOG_TAG_ACTIVITY, activity.getClass().getSimpleName() + "InstanceState");
    }

    @Override
    public void onActivityDestroyed(Activity activity) {
        AppLog.logDebug(AppLog.LOG_TAG_ACTIVITY, activity.getClass().getSimpleName() + "Destroyed");
    }
}

File: appinkotlin/src/main/kotlin/com/guoshujinfu/aqrlei/base/BaseApplication.kt
package com.guoshujinfu.aqrlei.base

import android.app.Activity
import android.app.Application
import android.os.Bundle
import com.facebook.drawee.backends.pipeline.Fresco

/**
 * @Author: AqrLei
 *@Name MyLearning
 *@Description:
 *@Date: 2017/8/17
 */
class BaseApplication: Application(), Application.ActivityLifecycleCallbacks {
    override fun onActivityPaused(p0: Activity?) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun onActivityResumed(p0: Activity?) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun onActivityStarted(p0: Activity?) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun onActivityDestroyed(p0: Activity?) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun onActivitySaveInstanceState(p0: Activity?, p1: Bundle?) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun onActivityStopped(p0: Activity?) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun onActivityCreated(p0: Activity?, p1: Bundle?) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun onCreate() {
        super.onCreate()
        registerActivityLifecycleCallbacks(this)
        Fresco.initialize(this)

    }
}